connect/runtime/src/main/java/org/apache/kafka/connect/storage/OffsetStorageReaderImpl.java

```diff
@@ -108,4 +140,14 @@ public OffsetStorageReaderImpl(OffsetBackingStore backingStore, String namespace
 
         return result;
     }
+
+    public void close() {
+        synchronized (offsetReadFutures) {
```

Comment:
We should first check to see if this is closed, and if so simply return.

Reply:
Ack, will address.